tests/logictest/http_connector.py

```diff
@@ -181,16 +185,20 @@ def query_with_session(self, statement):
         if self._session is None:
             if response is not None and "session_id" in response:
                 self._session = {"id": response["session_id"]}
-        return response
+        return response_list
 
     def fetch_all(self, statement):
-        # TODO use next_uri to get all results
-        resp = self.query_with_session(statement)
-        if resp is None:
+        resp_list = self.query_with_session(statement)
+        if len(resp_list) == 0 :
             log.warning("fetch all with empty results")
```

Comment:
sometimes it is expect to get empty results？ does it happen frequently in tests?

Reply:
add `after 6 secs` in the log to remind others it may be a slow/blocked query?

Reply:
I suggest :

1.  warning if next URL is not  None after 6 secs
2. interval from  2 secs to 1 secs or 0.5 secs to make the whole test faster?